src/sentry/snuba/metrics/query_builder.py

```diff
@@ -170,11 +176,9 @@ def __init__(self, query_params, paginator_kwargs: Optional[Dict] = None):
         if len(raw_fields) == 0:
             raise InvalidField('Request is missing a "field"')
 
-        self.fields = {}
+        self.fields = []
         for key in raw_fields:
-            op, metric_mri = parse_field(key)
-            mri_key = f"{op}({metric_mri})" if op is not None else metric_mri
-            self.fields[mri_key] = (op, metric_mri)
+            self.fields.append(parse_field(key, query_params))
 
         self.orderby = self._parse_orderby(query_params)
```

Comment:
Currently all validation happens in methods of `APIQueryDefinition`. I think it makes sense to keep it there in most cases, but we might want to move some validation to `QueryDefinition` or `get_series` if we think it needs to run regardless of the source of the request.